src/managementEnforcer.ts

```diff
@@ -300,6 +310,10 @@ export class ManagementEnforcer extends InternalEnforcer {
    * @return succeeds or not.
    */
   public async updateNamedPolicy(ptype: string, oldRule: string[], newRule: string[]): Promise<boolean> {
+    if (this.watcher && this.autoNotifyWatcher) {
+      // error intentionally ignored
```

Comment:
I think this should call remove and add policy. We cannot ignore calls.

Reply:
RN all the watcher alerts calls the loadPolicy  method, maybe we create a watcherEx method for this change.
but it is not ignored.

Reply:
Since `golang` has it, maybe we should too? https://github.com/JalinWang/casbin/blob/master/internal_api.go#L184

Reply:
the method used is empty https://github.com/JalinWang/casbin/blob/74e094778fb3b2c3ed2e7a32d51dbaba59c8df39/watcher_update_test.go#L25

Reply:
> RN all the watcher alerts calls the loadPolicy method, maybe we create a watcherEx method for this change.
but it is not ignored.

I don't think we should assume that user defines a normal watcher when they have watcherEx in place. IMO all functions should be covered, but I'll let @hsluoyz or @nodece to be the final judge in that.

Reply:
As said, `golang` covers all update functions.

Reply:
We need to wait for the adapter to end, and then call the watcher.

Reply:
@Shivansh-yadav13